### PR TITLE
added test cases infrastructure

### PR DESCRIPTION
**Pull request by boroda4436**

### CODE REVIEW

File: src/main/java/core/basesyntax/dao/impl/AbstractDao.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.dao.impl;
+
+import org.hibernate.SessionFactory;
+
+public abstract class AbstractDao {
+    protected final SessionFactory sf;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/impl/AbstractDao.java:6]:
```suggestion
    protected final SessionFactory factory;
```

File: src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            transaction = session.beginTransaction();
+            session.persist(smile);
+            transaction.commit();
+            return smile;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert Smile entity", e);
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java:31]:
```suggestion
        } finally {
            if (session != null) {
                session.close();
            }
        }
```

File: src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Override
+    public List<Smile> getAll() {
+        try (Session session = sf.openSession()) {
+            CriteriaQuery<Smile> criteriaQuery = session.getCriteriaBuilder()
+                    .createQuery(Smile.class);
+            criteriaQuery.from(Smile.class);
+            return session.createQuery(criteriaQuery).getResultList();
+        } catch (Exception e) {
+            return new ArrayList<>();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java:47]: maybe throw RuntimeException?

File: src/test/java/core/basesyntax/AbstractTest.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        Class<? extends DataSource> dataSourceClassName();
+
+        Properties dataSourceProperties();
+
+        List<IdentifierStrategy> identifierStrategies();
+
+        Database database();
+    }
+
+    private SessionFactory sf;
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/AbstractTest.java:40]:
```suggestion
    private SessionFactory factory;
```

File: src/test/java/core/basesyntax/AbstractTest.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    protected Properties getProperties() {
+        Properties properties = new Properties();
+        properties.put("hibernate.dialect", getDataSourceProvider().hibernateDialect());
+        properties.put("hibernate.hbm2ddl.auto", "create-drop");
+        //log settings
+        //properties.put("hibernate.show_sql", Boolean.TRUE.toString());
+        //properties.put("hibernate.format_sql", Boolean.TRUE.toString());
+        //properties.put("hibernate.use_sql_coments", Boolean.FALSE.toString());
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/AbstractTest.java:78]: do we need these?

**Review by KseniiaMakarova — APPROVED** [review]

File: src/test/java/core/basesyntax/AbstractTest.java (hunk 5 above)
Comment (boroda4436) [reply src/test/java/core/basesyntax/AbstractTest.java:78]: let's leave it for now.

File: src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java (hunk 2 above)
Comment (boroda4436) [reply src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java:36]: done, thanks

**Pull request merged**